app/api/currentuser/search_for_available_groups.feature

```diff
@@ -0,0 +1,155 @@
+      | 9  | Friends   | Some other friends                        | Another friends group  | 1           |
+      | 21 | UserSelf  | (the) user self                           |                        | 0           |
+    And the database has the following table 'groups_groups':
+      | ID | idGroupParent | idGroupChild | sType              | sStatusDate          |
+      | 2  | 1             | 21           | invitationSent     | 2017-02-29T06:38:38Z |
+      | 3  | 2             | 21           | invitationRefused  | 2017-03-29T06:38:38Z |
+      | 4  | 3             | 21           | requestSent        | 2017-04-29T06:38:38Z |
+      | 5  | 4             | 21           | requestRefused     | 2017-05-29T06:38:38Z |
+      | 6  | 5             | 21           | invitationAccepted | 2017-06-29T06:38:38Z |
+      | 7  | 6             | 21           | requestAccepted    | 2017-07-29T06:38:38Z |
+      | 8  | 7             | 21           | removed            | 2017-08-29T06:38:38Z |
+      | 9  | 8             | 21           | left               | 2017-09-29T06:38:38Z |
+      | 10 | 9             | 21           | direct             | 2017-10-29T06:38:38Z |
+      | 11 | 1             | 22           | direct             | 2017-11-29T06:38:38Z |
+
+  Scenario: Search for groups with "the"
+    Given I am the user with ID "1"
+    When I send a GET request to "/current-user/available-groups?search=the"
+    Then the response code should be 200
+    And the response body should be, in JSON:
+    """
+    [
+      {
+        "id": "2",
+        "name": "(the) Our Team ___",
+        "description": "Our team group",
+        "type": "Team"
+      },
+      {
+        "id": "4",
+        "name": "(the) |||Our Friends \\\\\\%\\\\%\\ :)",
+        "description": "Group for our friends",
+        "type": "Friends"
+      },
+      {
+        "id": "7",
+        "name": "Another %%%Team",
+        "description": "Another team group",
+        "type": "Team"
+      },
+      {
+        "id": "8",
+        "name": "Another %%%Club",
+        "description": "Another club group",
+        "type": "Club"
+      }
+    ]
+    """
+
+  Scenario: Search for groups with "the" (limit=2)
+    Given I am the user with ID "1"
+    When I send a GET request to "/current-user/available-groups?search=the&limit=2"
+    Then the response code should be 200
+    And the response body should be, in JSON:
+    """
+    [
+      {
+        "id": "2",
+        "name": "(the) Our Team ___",
+        "description": "Our team group",
+        "type": "Team"
+      },
+      {
+        "id": "4",
+        "name": "(the) |||Our Friends \\\\\\%\\\\%\\ :)",
+        "description": "Group for our friends",
+        "type": "Friends"
+      }
+    ]
+    """
+
+  Scenario: Search for groups with percent signs
+    Given I am the user with ID "1"
+    When I send a GET request to "/current-user/available-groups?search=%25%25%25"
```

Comment:
You should put a comment next on this line with the search you are doing so that the reader can understand the test immediately. (same below with the other encoded string searchs)